internal/config/generate/main.go

```diff
@@ -0,0 +1,133 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package main
+
+import (
+	"fmt"
+	"go/format"
+	"os"
+	"sort"
+	"strings"
+	"unicode"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+
+	"github.com/BurntSushi/toml"
+)
+
+// function names
+var functions = []string{}
+
+// A value from the toml global level map.
+type EnvGet struct {
+	Default *string `toml:"default"`
+	GoType  string  `toml:"go-type"`
+	Private bool    `toml:"private"`
+}
+
+func (e EnvGet) toFunction(env string) string {
+	name := toFunctionName(env)
+	typ := e.GoType
+	get := "get"
+
+	var defaultValue string
+	hasDefault := e.Default != nil
+	if hasDefault {
+		defaultValue = *e.Default
+	}
+
+	to_ := []rune(e.GoType)
+	to_[0] = unicode.ToUpper(to_[0])
+	to := "to" + string(to_)
+
+	args := fmt.Sprintf(`"%s", "%s", %t, %s`, env, defaultValue, hasDefault, to)
+
+	if e.Private {
+		name = "get" + name
+		typ = "*" + typ
+		get += "Optional"
+	} else {
+		name = "Get" + name
+		functions = append(functions, name)
+	}
+
+	return fmt.Sprintf("func %s() %s { return %s(%s) }\n", name, typ, get, args)
+}
+
+// Splits the string by "_" and joins each substring with the first letter in upper case.
+func toFunctionName(env string) string {
+	caser := cases.Title(language.English)
+	words := strings.Split(env, "_")
+	for i, word := range words {
+		words[i] = caser.String(word)
+	}
+	return strings.Join(words, "")
+}
+
+func addLine(builder *strings.Builder, s string, a ...any) {
+	builder.WriteString(fmt.Sprintf(s, a...))
+	builder.WriteString("\n")
+}
+
+func main() {
+	// reading the toml file
+	bytes, err := os.ReadFile("generate/Config.toml")
+	if err != nil {
+		panic(err)
+	}
+
+	// decoding the toml file
+	var config map[string]EnvGet
+	_, err = toml.Decode(string(bytes), &config)
+	if err != nil {
+		panic(err)
+	}
+
+	// creating the header
+	var builder strings.Builder
+	addLine(&builder, `// (c) Cartesi and individual authors (see AUTHORS)`)
+	addLine(&builder, `// SPDX-License-Identifier: Apache-2.0 (see LICENSE)`)
+
+	addLine(&builder, "")
+	addLine(&builder, `// Code generated by internal/config/generate.`)
+	addLine(&builder, `// DO NOT EDIT.`)
```

Comment:
If this is the first line of the file my editor opens the file as read-only. Would you change it?